### PR TITLE
update http and asto

### PR DESCRIPTION
**Pull request by Guseyn**
**The main point** of this pr is to update http and asto deps. NPM adapter has problems with blocking resources, which I am considering to fix in the another pr. For now I use two different contexts of vertex, so that tests would pass. But obviously this stuff needs to be improved.

#44

### CODE REVIEW

**Comment by Guseyn** [comment]
Update: I disabled tests because of window, I am going to fix it in another pr.

**Comment by 0crat** [comment]
This pull request [#43](https://github.com/artipie/npm-adapter/issues/43) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/34031508-898b-4b9e-8cf7-19b2b63c3618); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/npm/misc/DescSortedVersions.java [hunk 1]
```diff
@@ -86,7 +88,7 @@ private static int compareVersions(final String v1, final String v2) {
             result = Integer.valueOf(component1[index])
                 .compareTo(Integer.parseInt(component2[index]));
             if (result != 0) {
-                break;
+                return result;
```
Comment (Vatavuk) [inline src/main/java/com/artipie/npm/misc/DescSortedVersions.java:91]: @Guseyn why is this changed?

Comment (Guseyn) [reply src/main/java/com/artipie/npm/misc/DescSortedVersions.java:91]: @Vatavuk It's because we need to return value here otherwise it will be rewritten and we will get incorrect comparable number. I am considering also write another test on this method in another pr.

Comment (Vatavuk) [reply src/main/java/com/artipie/npm/misc/DescSortedVersions.java:91]: @Guseyn ok

File: src/main/java/com/artipie/npm/http/UploadSlice.java [hunk 2]
```diff
@@ -96,7 +97,9 @@ public Response response(
                             )
                         )
                     ),
-                    Flowable.fromPublisher(publisher)
+                    new Content.From(
+                        Flowable.fromPublisher(publisher)
```
Comment (g4s8) [inline src/main/java/com/artipie/npm/http/UploadSlice.java:101]: @Guseyn it's not needed to wrap a `Publisher` with a `Flowable`, since the constructor of `Content.From` accepts `Publisher`

File: src/test/java/com/artipie/npm/NpmCommandsTest.java [hunk 3]
```diff
@@ -47,19 +50,40 @@
...
+    private Vertx vertx;
+
+    @BeforeEach
+    void setUp() {
+        this.vertx = Vertx.vertx();
+    }
+
+    @AfterEach
+    void tearDown() {
+        if (this.vertx != null) {
```
Comment (g4s8) [inline src/test/java/com/artipie/npm/NpmCommandsTest.java:67]: @Guseyn why do we need this check? `this.vertx` instance should not be `null` in the test, because it's instanticated in `setUp()` method

File: src/test/java/com/artipie/npm/NpmCommandsTest.java [hunk 4]
```diff
@@ -47,19 +50,40 @@
...
+    void npmPublishAndInstallWorks()
         throws IOException, InterruptedException {
-        final Storage storage = new FileStorage(Files.createTempDirectory("temp"));
-        final Vertx vertx = Vertx.vertx();
-        final NpmRegistry registry = new NpmRegistry(vertx, storage);
+        final Storage storage = new FileStorage(
+            Files.createTempDirectory("temp"),
+            this.vertx.fileSystem()
+        );
+        final Vertx local = Vertx.vertx();
```
Comment (g4s8) [inline src/test/java/com/artipie/npm/NpmCommandsTest.java:85]: @Guseyn why do you use different `Vertx` instances here? I suppose you can just use `this.vertx`

File: src/test/java/com/artipie/npm/NpmCommandsTest.java [hunk 5]
```diff
@@ -79,14 +103,15 @@ public final void npmPublishAndInstallWorks()
...
     @Test
-    public final void npmInstallWithFilePrefixWorks()
+    @Disabled
+    void npmInstallWithFilePrefixWorks()
         throws IOException, InterruptedException {
         final Path temp = Files.createTempDirectory("temp");
-        final Storage storage = new FileStorage(temp);
-        final Vertx vertx = Vertx.vertx();
+        final Storage storage = new FileStorage(temp, this.vertx.fileSystem());
+        final Vertx local = Vertx.vertx();
```
Comment (g4s8) [inline src/test/java/com/artipie/npm/NpmCommandsTest.java:111]: @Guseyn same here: why another `Vertx` instance?

File: src/main/java/com/artipie/npm/http/UploadSlice.java (hunk 2 above)
Comment (Guseyn) [reply src/main/java/com/artipie/npm/http/UploadSlice.java:101]: @g4s8 okay I will fix it

File: src/test/java/com/artipie/npm/NpmCommandsTest.java (hunk 3 above)
Comment (Guseyn) [reply src/test/java/com/artipie/npm/NpmCommandsTest.java:67]: @g4s8 okay, will be fixed, I need to check why

File: src/test/java/com/artipie/npm/NpmCommandsTest.java (hunk 4 above)
Comment (Guseyn) [reply src/test/java/com/artipie/npm/NpmCommandsTest.java:85]: @g4s8 because I don't know now another solution, as I told in the meeting today tests should be rewritten otherwise we will get blocking exceptions. Something is wrong with implementation in asto, but I don't know for sure

File: src/test/java/com/artipie/npm/NpmCommandsTest.java (hunk 5 above)
Comment (Guseyn) [reply src/test/java/com/artipie/npm/NpmCommandsTest.java:111]: @g4s8 see above comment

File: src/test/java/com/artipie/npm/NpmCommandsTest.java (hunk 4 above)
Comment (Guseyn) [reply src/test/java/com/artipie/npm/NpmCommandsTest.java:83]: @g4s8 anyway tests with two vetrexes are disabled, they are needed to be rewritten, or/and maybe we could check asto usage of file system object.

**Comment by Guseyn** [comment]
@g4s8 plz check the updates and replies above

File: src/main/java/com/artipie/npm/http/UploadSlice.java (hunk 2 above)
Comment (Guseyn) [reply src/main/java/com/artipie/npm/http/UploadSlice.java]: @g4s8 fixed

File: src/test/java/com/artipie/npm/NpmCommandsTest.java (hunk 3 above)
Comment (Guseyn) [reply src/test/java/com/artipie/npm/NpmCommandsTest.java]: @g4s8 fixed

**Comment by Guseyn** [comment]
@g4s8 ping

**Comment by Guseyn** [comment]
@g4s8 ping

**Comment by Guseyn** [comment]
@g4s8 ping

**Comment by Guseyn** [comment]
@g4s8 ping

File: src/test/java/com/artipie/npm/NpmCommandsTest.java (hunk 4 above)
Comment (g4s8) [reply src/test/java/com/artipie/npm/NpmCommandsTest.java:83]: @Guseyn why did you disable these tests? I've just run it, and both tests passed

File: src/test/java/com/artipie/npm/NpmCommandsTest.java (hunk 5 above)
Comment (g4s8) [reply src/test/java/com/artipie/npm/NpmCommandsTest.java:109]: @Guseyn same here: this test is passing when not disabled

**Comment by Guseyn** [comment]
@g4s8 The tests you mentioned fail on windows as I've told in a meeting before.

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@Vatavuk please make sure you will find at least 3 issues during next CR, thanks

@Guseyn please make sure you start all your comments with the name of the user they are referred to, [see](https://github.com/artipie/npm-adapter/pull/43#issuecomment-612981271)

**Comment by Vatavuk** [comment]
@sereshqua ok

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
> @0crat quality acceptable ([here](https://github.com/artipie/npm-adapter/issues/43#issuecomment-615332986))

@sereshqua There is no quality review for [#43](https://github.com/artipie/npm-adapter/issues/43), no performer

**Comment by Vatavuk** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/artipie/npm-adapter/issues/43#issuecomment-615372265))

@Vatavuk This is what I know about this job in CT2E6TK9B, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#43](https://github.com/artipie/npm-adapter/issues/43) is not in scope
  * Job [footprint](https://www.0crat.com/footprint/CT2E6TK9B?q=%7Bjob%3A%27gh%3Aartipie%2Fnpm-adapter%2343%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)
